File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.cpp
class Solution {
public:
    int minGroups(vector<vector<int>>& intervals) {
        vector<int> sorted_start, sorted_end;

        for (const auto& interval: intervals) {
            sorted_start.push_back(interval[0]);
            sorted_end.push_back(interval[1]);
        }

        sort(sorted_start.begin(), sorted_start.end());
        sort(sorted_end.begin(), sorted_end.end());

        int result = 0, idx = 0;

        for (int start: sorted_start) {
            if (start > sorted_end[idx]) {
                idx++;
            } else {
                result++;
            }
        }

        return result;
    }
};

File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.java
class Solution {
    public int minGroups(int[][] intervals) {
        int[] sortedStart = new int[intervals.length];
        int[] sortedEnd = new int[intervals.length];

        for (int i = 0; i < intervals.length; i++) {
            sortedStart[i] = intervals[i][0];
            sortedEnd[i] = intervals[i][1];
        }

        Arrays.sort(sortedStart);
        Arrays.sort(sortedEnd);

        int result = 0, idx = 0;

        for (int start: sortedStart) {
            if (start > sortedEnd[idx]) {
                idx++;
            } else {
                result++;
            }
        }

        return result;
    }
}